Projects/Core/Services/SnapshotClient/SnapshotClient.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  SnapshotClient.swift
+//  Services
+//
+//  Created by GREEN on 6/5/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import ComposableArchitecture
+import SwiftUI
+
+// MARK: - API Client Interface
+@DependencyClient
+public struct SnapshotClient {
+  var takeSnapshot: @Sendable () async throws -> UIImage
+}
+
+public extension DependencyValues {
+  var snapshotClient: SnapshotClient {
+    self[SnapshotClient.self]
+  }
+}
+
+// MARK: - API Client Implementation
+extension SnapshotClient: DependencyKey {
+  public static let liveValue = SnapshotClient(
+    takeSnapshot: {
+      try await MainActor.run {
+        var totalImage: UIImage?
+        
+        let keyWindow = UIApplication.shared.connectedScenes
+          .compactMap { $0 as? UIWindowScene }
+          .flatMap { $0.windows }
+          .first { $0.isKeyWindow }
+        guard let currentLayer = keyWindow?.layer else {
+          throw SnapshotError(code: .failedToGetKeyWindowLayer)
+        }
+        
+        let currentScale = UIScreen.main.scale
+        UIGraphicsBeginImageContextWithOptions(currentLayer.frame.size, false, currentScale)
+        
+        guard let currentContext = UIGraphicsGetCurrentContext() else {
+          throw SnapshotError(code: .failedToGetCurrentContext)
+        }
+        currentLayer.render(in: currentContext)
+        
+        totalImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return totalImage ?? UIImage()
+      }
+    }
```

Comment:
P5
```suggestion
takeSnapshot: {
      var totalImage: UIImage?
      
      let keyWindow = await MainActor.run {
        UIApplication.shared.connectedScenes
          .compactMap { $0 as? UIWindowScene }
          .flatMap { $0.windows }
          .first { $0.isKeyWindow }
      }
      
      guard let currentLayer = await keyWindow?.layer else {
        throw SnapshotError(code: .failedToGetKeyWindowLayer)
      }
      
      let currentScale = await UIScreen.main.scale
      UIGraphicsBeginImageContextWithOptions(currentLayer.frame.size, false, currentScale)
      
      guard let currentContext = UIGraphicsGetCurrentContext() else {
        throw SnapshotError(code: .failedToGetCurrentContext)
      }
      currentLayer.render(in: currentContext)
      
      totalImage = UIGraphicsGetImageFromCurrentImageContext()
      UIGraphicsEndImageContext()
      return totalImage ?? UIImage()
    }
```
MainActor를 완전히 감싸지 않는 방법도 있습니다. 이런 방법이 있다는 정도로만 알고 계셔도 좋습니다.